core/src/com/google/inject/spi/BindingSourceRestriction.java

```diff
@@ -345,8 +345,12 @@ private static Stream<Class<? extends Annotation>> getPermits(Class<?> clazz) {
           Stream.concat(
               annotations, Arrays.stream(clazz.getAnnotatedSuperclass().getAnnotations()));
     }
-    return annotations
```

Comment:
Why change?

Reply:
I kept getting a build error here from maven

```
Type mismatch: cannot convert from Stream<Class<capture#26-of ? extends Annotation>> to Stream<Class<? extends Annotation>>Java(16777235)
```

Introducing this expliocit cast resolved it, so i guess i cannot figure out why this isn't a problem at head?

im guessing it is because i am using a different version of `javac` 

```
% javac --version
javac 21.0.6
```

Reply:
ok, i was confused before. this appears to be an incompatibiltiy with `ecj` which is used by the vscode java plugins... sigh.  It looks like there are number of issues where ecj reports and error and javac does not (and visa versa).

Added a comment to that effect.  I can revert if you prefer